src/Utilities/Extensions/ITypeSymbolExtensions.cs

```diff
@@ -103,6 +105,14 @@ public static bool DerivesFrom(this ITypeSymbol symbol, ITypeSymbol candidateBas
             return false;
         }
 
+        public static bool IsAssignableTo(this ITypeSymbol fromSymbol, ITypeSymbol toSymbol, Compilation compilation)
+        {
+            return fromSymbol != null
+                && toSymbol != null
+                && ((compilation is CSharpCompilation && ((CSharpCompilation)compilation).ClassifyConversion(fromSymbol, toSymbol).IsImplicit)
```

Comment:
Please don't add types from language specific code analysis assemblies into the core utilities. This will cause a C# analyzer to unnecessarily load VB CodeAnalysis assembly and vice versa. This is the primary reason for the current implementation being cloned in both language specific analyzers.

We can make a change similar to the one you are proposing once Roslyn exposes conversion information at the language agnostic layer.

Reply:
Makes sense. How would I go about 333fred's request to move IsAssignableTo into an extension? Should I instead duplicate IsAssignableTo code and move it into the DoNotRaiseExceptionInUnexpectedLocations?

Reply:
You can either keep the current strategy of cloning the code into each analyzer OR create two new shared projects CSharpAnalyzerUtilities and BasicAnalyzerUtilities and put language specific helpers there.